js/eventloop.go

```diff
@@ -0,0 +1,155 @@
+package js
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/dop251/goja"
+)
+
+// an event loop
+type eventLoop struct {
+	lock          sync.Mutex
+	queue         []func() error
```

Comment:
I can move `func() error` in a type ... but then it will need to be cast in some places where I would want to reference `js` package from a module. 

But otherwise the `queue` and `reservedCount` are super connected and used almost always together to the point that I will need to keep them together and then that is most of the event loop, so I don't see any benefit. Maybe in the future when the event loop grows it can be more useful :shrug: